lib/SILOptimizer/Utils/OwnershipOptUtils.cpp

```diff
@@ -244,14 +162,39 @@ static bool canFixUpOwnershipForRAUW(SILValue oldValue, SILValue newValue,
   if (!hasValidRAUWOwnership(oldValue, newValue))
     return false;
 
-  if (oldValue.getOwnershipKind() == OwnershipKind::Guaranteed) {
-    // Check that the old lifetime can be extended and record the necessary
-    // book-keeping in the OwnershipFixupContext.
-    if (!findTransitiveBorrowedUses(oldValue, context.transitiveBorrowedUses,
-                                    context.recursiveReborrows)) {
-      context.clear();
-      return false;
-    }
+  if (oldValue.getOwnershipKind() != OwnershipKind::Guaranteed)
+    return true;
+
+  // Check that the old lifetime can be extended and record the necessary
+  // book-keeping in the OwnershipFixupContext.
+  context.clear();
+
+  // FIXME: the rest of the logic in this function a shim between OwnershipUtils
+  // and OwnershipContextFixup that will be simplified in a follow-up commit.
```

Comment:
Can you put in a comment here something that explains that we are implementing the same thing as the extended lifetime version but we want to have more control. Even if this is going away, when I was reading the code, the way it is laid out made me look for bugs there. So a more explicit comment connecting the two together would be appreciated.